### DISCUSSION TITLE
How do I add authentication to a listener (TLS enabled) without downtime for clients?

### DISCUSSION THREAD

**Discussion by ghost** [strimzi/strimzi-kafka-operator#8364] [Q&A] [answered]
I have an external listener with TLS enabled. I want to add authentication (scram-sha-512) to it, but without any downtime for the clients.

One way to do this is to add a separate listener with authentication enabled, and then upgrade the clients one by one, and deprecate the old listener. Is there any other option at my disposal?

**Comment by scholzj** [comment, marked as answer]
I think that is the best option to do it.
